iocore/aio/AIO.cc

```diff
@@ -379,18 +379,15 @@ aio_queue_req(AIOCallbackInternal *op, int fromAPI = 0)
 static inline int
 cache_op(AIOCallbackInternal *op)
 {
-  bool read = (op->aiocb.aio_lio_opcode == LIO_READ);
+  typedef ssize_t (*p)(int, void*, size_t, __off_t);
```

Comment:
No new `typedef` usage allowed. `using` is the new standard (see [here](https://en.cppreference.com/w/cpp/language/using_declaration)). E.g.
```
using p = ssize_t(int, void*, size_t, __off_t);
```
Even better, though, would using `decltype` and a name more descriptive than "p". Because the point of the cast isn't obvious, a comment is useful. This uses `reinterpret_cast` because we no longer allow C style casts.
```
  using ActionFunc = decltype(&pread);
  // Need to cast @c pwrite because it takes a const void * vs. a void * for @c pread.
  auto action = op->aiocb.aio_lio_opcode == LIO_READ ? &pread : reinterpret_cast<ActionFunc>(&pwrite);
```